### ISSUE TITLE
Plugin parameter configurations do not support more than 3 layers of nesting

### ISSUE THREAD

**Issue by ccjinghang** [Kong/kong#4672]
Hello!
My schema file is as follows:

```
local typedefs = require "kong.db.schema.typedefs"

local child_elements_contion = {
  type = "record",
  fields = {
    { mark_key = { type = "string" }, },
	{ mark_key_index = { type = "number", default = 1 }, },
	{ mark_key_value = { type = "string" }, },
	{ mark_type = { type = "string" ,  one_of = { "path", "query", "header", "body" },}, },
	{ mark_method = { type = "string",one_of = { "start_with", "end_with", "contain", "regex" }, }, },
  },
 }
			  
local child_child_elements_relation = {
  type = "record",
  --required = true,
  fields = {
    { condition = { type = "array", default = {}, elements = child_elements_contion, }, },
    { relationship = { type = "string" , default = "AND" ,one_of = { "AND", "NOR", "NOT" },}, },
  },
}		


local child_elements_relation = {
  type = "record",
  --required = true,
  fields = {
    { condition = { type = "array", default = {}, elements = child_elements_contion, }, },
    { ralation = { type = "array", default = {}, elements = child_child_elements_relation, }, },
    { relationship = { type = "string" , default = "AND" ,one_of = { "AND", "NOR", "NOT" },}, },
  },
}				  
			  
local elements_record = {
  type = "record",
  fields = { 
    { condition = { type = "array", default = {}, elements = child_elements_contion, }, },
    { ralation = { type = "array", default = {}, elements = child_elements_relation, }, },
    { relationship = { type = "string" , default = "AND" ,one_of = { "AND", "NOR", "NOT" },}, },
	{ ip = { type = "string" }, },
	{ port = { type = "number"}, },
	{ upstream = { type = "string" }, },
  },
}
			
local mark_array = {
  type = "array",
  default = {},
  elements = elements_record,
}

return {
  name = "mark-route",
  fields = {
    { config = {
        type = "record",
        fields = {
          { mark_number = { type = "number", default = 0 }, },
          { mark = mark_array,  },
		  { timestamp = { type = "number" ,default = 0}, },
        },
    }, },
  },
}
```

**An error occurred when I set the parameters as follows:**
```
config.mark[1].ralation[1].condition[1].mark_key = 3
```

**Error is as follows:**
```
2019/05/31 14:19:56 [error] 6350#0: *12665 [lua] init.lua:161: handle_error(): /usr/local/share/lua/5.1/lapis/request.lua:291: attempt to index local 'curr' (a string value)
```

Can you help me solve this problem?

**Comment by ccjinghang** [comment]
Success when I set the parameters as follows:
config.mark[1].condition[1].mark_key = 3

**Is it not supported to parse more than 3 levels of nesting?**

**Comment by bungle** [comment]
It looks like the Lapis bug I reported in January 2018 ;-):
https://github.com/leafo/lapis/issues/598

Or perhaps a new error related to it. Let me check if I can upgrade our Lapis dependency as that is fixed in `1.7.0`.

**Comment by ccjinghang** [comment]
Thank you for your reply and look forward to having this problem solved

**Comment by kikito** [comment]
Hi @ccjinghang,

We recently released Kong 1.3.0rc1 which contains a possible fix for your issue (note that @bungle was not able to reproduce it even before the upgrade, though).

Could you please try your plugin in 1.3.0rc1?

**Comment by ccjinghang** [comment]
Sorry,late reply.
After my verification, the bug was fixed. @kikito
